src/pages/workspaces/workspace/DeleteWorkspaceModal.js

```diff
@@ -1,23 +1,67 @@
 import _ from 'lodash/fp'
 import { useState } from 'react'
-import { div, h, label, span } from 'react-hyperscript-helpers'
+import { div, h, label, p, span } from 'react-hyperscript-helpers'
 import { ButtonPrimary, Link, spinnerOverlay } from 'src/components/common'
+import { warningBoxStyle } from 'src/components/data/data-utils'
+import { icon } from 'src/components/icons'
 import { TextInput } from 'src/components/input'
 import Modal from 'src/components/Modal'
 import { Ajax } from 'src/libs/ajax'
-import { bucketBrowserUrl } from 'src/libs/auth'
+import { bucketBrowserUrl, getUser } from 'src/libs/auth'
+import colors from 'src/libs/colors'
 import { reportError } from 'src/libs/error'
+import { isAppDeletable } from 'src/libs/runtime-utils'
 import * as Utils from 'src/libs/utils'
 
 
 const DeleteWorkspaceModal = ({ workspace: { workspace: { namespace, name, bucketName } }, onDismiss, onSuccess }) => {
   const [deleting, setDeleting] = useState(false)
   const [deleteConfirmation, setDeleteConfirmation] = useState('')
+  const [loadingApps, setLoadingApps] = useState(true)
+  const [deletableApps, setDeletableApps] = useState()
+  const [nonDeletableApps, setNonDeletableApps] = useState()
+
+  const signal = Utils.useCancellation()
+
+  const loadApps = async workspaceName => {
+    const [currentWorkspaceAppList] = await Promise.all([
+      Ajax(signal).Apps.listWithoutProject({ creator: getUser().email, saturnWorkspaceName: workspaceName })
+    ])
+    const appPartition = _.partition(isAppDeletable, currentWorkspaceAppList)
+    setDeletableApps(appPartition[0])
+    setNonDeletableApps(appPartition[1])
+    setLoadingApps(false)
+  }
+
+  Utils.useOnMount(() => {
+    loadApps()
+  })
+
+  const getAppDeletionMessage = (deletableApps, nonDeletableApps) => {
+    return nonDeletableApps.length > 0 ?
```

Comment:
Generally, when you're checking if it's empty, it may be simpler to use `_.isEmpty`, for example here:
```suggestion
    return !_.isEmpty(nonDeletableApps) ?
```